### PR TITLE
Implement eventer_add_asynch_dep(...)

### PR DESCRIPTION
**Pull request by postwait**
This new function can be called from an asynchronous job to
delay the completion of the current executing job until the
new added deap has past its cleanup phase.

### CODE REVIEW

**Comment by aidancully** [comment]
I've got an alternative approach for this use case, I'll try to PR that, and we can compare... I think it wouldn't be hard to create a fork/join implementation that can be used with jobq's (like this does), or from other threads..

**Comment by postwait** [comment]
it would be nice to get some action on this.. I've brought this up to current master.

File: src/eventer/eventer_impl.c [hunk 1]
```diff
@@ -777,7 +777,29 @@ void eventer_add_asynch(eventer_jobq_t *q, eventer_t e) {
...
+  if(e->whence.tv_sec) {
+    job->timeout_event = eventer_alloc();
+    job->timeout_event->thr_owner = e->thr_owner;
+    memcpy(&job->timeout_event->whence, &e->whence, sizeof(e->whence));
+    job->timeout_event->mask = EVENTER_TIMER;
+    job->timeout_event->closure = job;
+    job->timeout_event->callback = eventer_jobq_execute_timeout;
+    eventer_add(job->timeout_event);
+  }
+  eventer_jobq_enqueue(q ? q : __default_jobq, job, eventer_jobq_inflight());
```
Comment (rileyberton) [inline src/eventer/eventer_impl.c]: `eventer_jobq_enqueue(job->jobq, ...);` eliminates the second branch here.

**Comment by rileyberton** [comment]
Looks good to me

**Pull request merged**